File: bank_server/src/main/java/edu/epitech/bank_server/domain/entities/BankCard.java
package edu.epitech.bank_server.domain.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "bank_cards")
public class BankCard {
    @Id
    private String accountNumber;
    private String cardholderName;
    private String cardSecurityCode;
    private String brandMark;
    private Date expirationDate;
}

File: terminal_app/lib/features/domain/entities/bank_card.dart

import 'package:equatable/equatable.dart';

class BankCard extends Equatable {
  final String accountNumber;
  final String cardholderName;
  final String cardSecurityCode;
  final String brandMark;
  final String expirationDate;

  const BankCard({
    required this.accountNumber,
    required this.cardholderName,
    required this.cardSecurityCode,
    required this.brandMark,
    required this.expirationDate
  });

  BankCard.fromJson(Map<String, dynamic> json)
    : accountNumber = json['accountNumber'],
      cardholderName = json['cardholderName'],
      cardSecurityCode = json['cardSecurityCode'],
      brandMark = json['brandMark'],
      expirationDate = json['expirationDate'];

  @override
  List<Object?> get props => [accountNumber, cardholderName, cardSecurityCode, brandMark, expirationDate];
}